src/modules/jcms_article_fragment/src/Form/ArticleFragmentForm.php

```diff
@@ -0,0 +1,109 @@
+ *
+ * @ingroup jcms_article_fragment
+ */
+class ArticleFragmentForm extends ContentEntityForm {
+
+  /**
+   * The entity manager.
+   *
+   * @var \Drupal\Core\Entity\EntityManagerInterface
+   */
+  protected $entityManager;
+
+  protected $api;
+
+  /**
+   * Constructs a ContentEntityForm object.
+   *
+   * @param \Drupal\Core\Entity\EntityManagerInterface $entity_manager
+   *   The entity manager.
+   */
+  public function __construct(EntityManagerInterface $entity_manager, FragmentApi $api) {
+    $this->entityManager = $entity_manager;
+    $this->api = $api;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static($container->get('entity.manager'), $container->get('jcms_article_fragment.api'));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    /* @var $entity \Drupal\jcms_article_fragment\Entity\ArticleFragment */
+    $form = parent::buildForm($form, $form_state);
+    if (!$this->entity->isNew()) {
+      $form['new_revision'] = [
+        '#type' => 'checkbox',
+        '#title' => $this->t('Create new revision'),
+        '#default_value' => FALSE,
+        '#weight' => 10,
+      ];
+    }
+    $entity = $this->entity;
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function save(array $form, FormStateInterface $form_state) {
+    $entity = &$this->entity;
+    // Save as a new revision if requested to do so.
+    if (!$form_state->isValueEmpty('new_revision') && $form_state->getValue('new_revision') != FALSE) {
+      $entity->setNewRevision();
+      // If a new revision is created, save the current user as revision author.
+      $entity->setRevisionCreationTime(REQUEST_TIME);
+      $entity->setRevisionUserId(\Drupal::currentUser()->id());
+    }
+    else {
+      $entity->setNewRevision(FALSE);
+    }
+    $status = parent::save($form, $form_state);
+    // Post image fragment to Lax.
+    $this->setImageFragment($form_state);
+    switch ($status) {
+      case SAVED_NEW:
+        drupal_set_message($this->t('Created the %label Article fragment.', [
+          '%label' => $entity->label(),
+        ]));
+        break;
+      default:
+        drupal_set_message($this->t('Saved the %label Article fragment.', [
+          '%label' => $entity->label(),
+        ]));
+    }
+    $form_state->setRedirect('entity.article_fragment.canonical', ['article_fragment' => $entity->id()]);
+  }
+
+  /**
+   * Sets the image for an article via an image fragment.
+   *
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   */
+  public function setImageFragment(FormStateInterface $form_state) {
+    $values = $form_state->getValues();
+    $article_id = $values['name'][0]['value'] ?? '';
+    $fid = $values['image'][0]['fids'][0] ?? 0;
+    if ($fid && $article_id) {
+      $this->api->postImageFragment($fid, $article_id);
+    }
```

Comment:
`postImageFragment()` returns a response; it must be checked so that if it's not 200 an error should be shown. Anything different from 200 means the fragment is in this intermediate state in which is stored in journal-cms, but has not been sent to Lax and hence it won't appear on the article page. There should be a way to distinguish between a fragment that has been successfully set on Lax and one that isn't.

Given we are working with remote procedure calls and Lax is the owner of the article data, maybe we should switch around the order: first store in Lax, only if successful there then store it on journal-cms; any error in Lax interrupts the whole flow.